pyvo/registry/tests/test_regtap.py

```diff
@@ -630,8 +646,12 @@ def test_ambiguous_access_url_warns(self, recwarn):
             "intf_roles": ["std"] * 2,
         })
         assert rsc.access_url == "http://a"
-        assert [str(w.message)[:50] for w in recwarn
-                ] == ['The resource ivo://pyvo/test_regtap.py has multipl']
+        # the following test is written this oddly to avoid that extra
+        # warnings break it.
+        expectation = set(
+            ['The resource ivo://pyvo/test_regtap.py has multipl'])
+        assert set(str(w.message)[:50] for w in recwarn
+            ) & expectation == expectation
```

Comment:
Isn't this a simpler way to check for the expected warning in the set?

```suggestion
        assert 'The resource ivo://pyvo/test_regtap.py has multipl' in [str(w.message)[:50] for w in recwarn]
```

Reply:
This feels a bit over engineered, any reasons not to use the `pytest.warns` context in tests?